packages/shared/src/component-utils.ts

```diff
@@ -2,21 +2,25 @@ import type { CSSProperties, ReactNode } from "react";
 import type { JSX } from "solid-js/jsx-runtime";
 import type { RenderFunction, StyleValue } from "vue";
 
+import { mimeTypes } from "@uploadthing/mime-types";
+
 import type { ExpandedRouteConfig } from "./types";
 import { objectKeys } from "./utils";
 
 export const generateMimeTypes = (fileTypes: string[]) => {
-  const accepted = fileTypes.map((type) => {
-    if (type === "blob") return "blob";
+  if (fileTypes.includes("blob")) return [];
+
+  return fileTypes.map((type) => {
     if (type === "pdf") return "application/pdf";
     if (type.includes("/")) return type;
-    else return `${type}/*`;
+    return [
+      // Add wildcard to support all subtypes, e.g. image => "image/*"
+      `${type}/*`,
+      // But some browsers/OSes don't support it, so we'll also dump all the mime types
+      // we know that starts with the type, e.g. image => "image/png, image/jpeg, ..."
+      ...objectKeys(mimeTypes).filter((t) => t.startsWith(type)),
+    ].join(", ");
   });
-
-  if (accepted.includes("blob")) {
```

Comment:
I think this case is still needed?

Reply:
nvm, I see it moved up, missed that at first

Reply:
i early-returned this case first line in the function